api/bibliotheca.py

```diff
@@ -1323,41 +1325,28 @@ def slice_timespan(self, start, cutoff, increment):
             yield slice_start, slice_cutoff, full_slice
             slice_start = slice_start + increment
 
-    def run_once(self, start, cutoff):
+    def catch_up_from(self, start, cutoff, progress):
         added_books = 0
         i = 0
         one_day = timedelta(days=1)
-        most_recent_timestamp = start
-        for start, cutoff, full_slice in self.slice_timespan(
-                start, cutoff, one_day):
-            most_recent_timestamp = start
-            self.log.info("Asking for events between %r and %r", start, cutoff)
-            try:
-                event = None
-                events = self.api.get_events_between(start, cutoff, full_slice)
-                for event in events:
-                    event_timestamp = self.handle_event(*event)
-                    if (not most_recent_timestamp or
-                        (event_timestamp > most_recent_timestamp)):
-                        most_recent_timestamp = event_timestamp
-                    i += 1
-                    if not i % 1000:
-                        self._db.commit()
-                self._db.commit()
-            except Exception, e:
-                if event:
-                    self.log.error(
-                        "Fatal error processing Bibliotheca event %r.", event,
-                        exc_info=e
-                    )
-                else:
-                    self.log.error(
-                        "Fatal error getting list of Bibliotheca events.",
-                        exc_info=e
```

Comment:
Also, exception handling is now done by `run_once`, so there's no need to have it in this method.